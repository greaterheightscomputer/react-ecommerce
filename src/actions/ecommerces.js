import uuid from 'uuid';

//Add_Ecommerce
export const addEcommerce = ({ description='', category='', item='', amount=0, image='', imageUrl='', stock=0, createdAt=0 } = {}) => ({
    type: 'ADD_ECOMMERCE',
    ecommerce: {
        id: uuid(),        
        description,
        category,
        item,
        amount,
        image,
        imageUrl,
        stock,
        createdAt
    }
});
// REMOVE_Ecommerce    
export const removeEcommerce = ( { id } = {}) => ({
    type: 'REMOVE_ECOMMERCE',
    id
});
// EDIT_Ecommerce  
export const editEcommerce = (id, updates) => ({
    type: 'EDIT_ECOMMERCE',
    id,
    updates
});
