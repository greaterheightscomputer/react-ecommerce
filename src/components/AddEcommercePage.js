import React from 'react';

const AddEcommercePage = () => (
    <div>
        This is from my add expense component
    </div>
);

export default AddEcommercePage;