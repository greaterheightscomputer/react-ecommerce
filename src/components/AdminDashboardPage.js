import React from 'react';
import AdminList from './AdminList';
import AdminListFilters from './AdminListFilters';
import EcommerceSummary from './EcommerceSummary';

const AdminDashboardPage = () => (
    <div>        
        <EcommerceSummary />
        <AdminListFilters />
        <AdminList />        
    </div>
);

export default AdminDashboardPage;