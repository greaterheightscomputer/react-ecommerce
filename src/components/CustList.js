import React from 'react';
import { connect } from 'react-redux';
import CustListItem from './CustListItem';
import selectEcommerce from '../selectors/ecommerces';

const CustList = (props) => (
    <div>
        <h1>Customer List</h1>
        {props.ecommerces.map((ecommerce) =>{
            return <CustListItem key={ecommerce.id} {...ecommerce} />                             
        })}      
       
    </div>
);

const mapStateToProps = (state) => {
    return {
        ecommerces: selectEcommerce(state.ecommerces, state.filters)        
    };
}

export default connect(mapStateToProps)(CustList);