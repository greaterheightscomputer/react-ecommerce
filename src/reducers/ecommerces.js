const ecommerceReducerDefaultState = [];
const ecommerceReducer = (state = ecommerceReducerDefaultState, action) => {
    switch (action.type) {
        case 'ADD_ECOMMERCE':            
            return [...state, action.ecommerce];   
        case 'REMOVE_ECOMMERCE':            
            return state.filter(({ id }) => id !== action.id); 
        case 'EDIT_ECOMMERCE':
            return state.map((ecommerce) => {
                if (ecommerce.id === action.id){
                    return {...ecommerce, ...action.updates};
                }else {
                    return ecommerce;
                }
            });              
        default: return state;        
    }
};

export default ecommerceReducer;