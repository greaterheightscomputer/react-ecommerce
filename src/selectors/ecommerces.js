import moment from "moment";

//Get visible ecommerce -> this function will match the ecommerce array object with the filters object
export default (ecommerces, { text, sortBy, itemType,  categoryType, startDate, endDate }) => {
    return ecommerces.filter((ecommerce) => {  //filter() allow us to return a subset of all of the array expenses.
        // const startDateMatch = typeof startDate !== 'number' || ecommerce.createdAt >= startDate;
        // const endDateMatch = typeof endDate !== 'number' || ecommerce.createdAt <= endDate;
        const createdAtMoment = moment(ecommerce.createdAt);
        const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment, 'day') : true;
        const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment, 'day') : true;
        const textMatch = ecommerce.description.toLowerCase().includes(text.toLowerCase());
        const categoryMatch = ecommerce.category.includes(categoryType);
        const itemMatch = ecommerce.item.includes(itemType);
        
        return startDateMatch && endDateMatch && textMatch && categoryMatch && itemMatch;
    }).sort((a, b) =>{    //compare function. sort() get called on a return subset array. if a < b return -1 else if a > b return 1  else a === b return 0
          if(sortBy === 'date') {
              return a.createdAt < b.createdAt ? 1 : -1;   // the truthy value is 1 which means b is more recent than a 
        }                                                  
        if(sortBy === 'amount') {
            return a.amount < b.amount ? 1 : -1;
      }
    });
};

