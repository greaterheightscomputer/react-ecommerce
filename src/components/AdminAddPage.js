import React from 'react';
import { connect } from 'react-redux';
import AdminForm from './AdminForm';
import { startAddEcommerce } from '../actions/ecommerces';

export class AdminAddPage extends React.Component {
    onSubmit = (ecommerce) => {
        // console.log(ecommerce)
        // props.dispatch(addEcommerce(ecommerce));
        this.props.startAddEcommerce(ecommerce);
        this.props.history.push('/admin_dashboard');
    };
    render() {
        return (
            <div className="list-body">
                <div className="page-admin-header">
                    <div className="content-container">
                        <h1 className="page-admin-header__title"> Add Product</h1>
                    </div>
                </div>                
                <div className="content-container">
                    <AdminForm
                        onSubmit={this.onSubmit}
                    />
                </div>
            </div>
        );
    };
};

const mapDispatchToProps = (dispatch) => ({
    startAddEcommerce: (ecommerce) => dispatch(startAddEcommerce(ecommerce))
});

export default connect(undefined, mapDispatchToProps)(AdminAddPage);