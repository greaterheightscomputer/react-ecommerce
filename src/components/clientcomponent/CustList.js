import React, { Component } from 'react';
import { connect } from 'react-redux';
import { ProductConsumer } from './context';
import CustListItem from '../../components/clientcomponent/CustListItem';
import CustTitle from '../../components/clientcomponent/CustTitle';
import { ecommerces, detailProduct } from '../clientcomponent/data';
import { addToCartAction } from '../../actions/cart';
import selectEcommerce from '../../selectors/ecommerces';

export  class CustList extends Component {    
    
    state = {        
        products: [],        
        detailProduct: detailProduct,
        cart: [{
            id:'-LynLz9jYS05Qm5pktMQ',
            description: 'Japan Shoe', 
            category: 'Men', 
            item: 'Men Footwear', 
            amount: 4150000, 
            image: 'shoe.png', 
            imageUrl: '../images/men_shoe1.jpg', 
            stock: 10, 
            company: 'Japan Company Ltd.',
            info:'The product is made for men who love and desirer of good quantity foot wear product. Our product has been in exist for 50 years now with the same high quantity materials and products. We have our customers global continuely demanding for our high product. The shipment of our product to any part of the world behind 24 hours of purchaseThe product is made for men who love and desirer of good quantity foot wear product. Our product has been in exist for 50 years now with the same high quantity materials and products. We have our customers global continuely demanding for our high product. The shipment of our product to any part of the world behind 24 hours of purchase',
            inCart: false,
            count: 0,
            total: 0,
            createdAt:1000
        }],
        cartSubTotal: 0,
        cartTax: 0,
        cartTotal: 0        
    };
    componentDidMount(){
        this.setProducts();        
    }            
    setProducts = () => {        
        let tempProducts = [];               
        // tempProducts = this.props.ecommerces;                  
        tempProducts = ecommerces;                         
        this.setState(() => {
            return { products: tempProducts };
        });
    };    
    
    addToCart = (product)=>{
     // console.log("clicked on Product: ", product);
        // let tempProducts = [...this.props.ecommerces]  
        let tempProducts = [...ecommerces]  
        const productIndex = tempProducts.findIndex(p => p.id === product.id);        
        product = tempProducts[productIndex];        
        // product.inCart=true;  
        product.count=1;
        const price = product.amount;
        product.total=price; 

        console.log(product);
        this.props.addToCartAction(product);                    
    };      
        
    render() {                  
        
        return (                    
            <div className="content-container">
            <ul>
                {/*
                    // this.state.cart.map(c => 
                    this.props.cart.map(c => 
                        <li>{c.description} | Units {c.count} | inCart:{c.inCart} | Price:{c.amount}
                        | Total:{c.amount}
                        </li>)                        
                */}
            </ul>
                <CustTitle name="our" title="products" />
                <div className="list-body">
                    <div className="list-items">
                    <ProductConsumer>
                        {value =>                                                                      
                            // this.props.ecommerces.length === 0 ? (
                            ecommerces.length === 0 ? (                                
                                // value.products.length === 0 ? (        
                                <div className="list-item list-item--message">
                                    <span>No Products</span>
                                </div>
                            ) : 
                                (
                                // this.props.ecommerces.map((product) =>{
                                ecommerces.map((product) =>{
                                    // value.products.map((product) =>{
                                    return <CustListItem
                                            key={product.id} product={product}                                            
                                            addToCart={this.addToCart}                                                                                        
                                />                                                                                                                 
                            }))                                                            
                        }
                    </ProductConsumer>                                                                         
                    </div>                                   
                </div>                           
            </div>                         
        )                
    }
}

const mapStateToProps = (state) => {            
    return {                        
        ecommerces: selectEcommerce(state.ecommerces, state.filters), 
        cart: state.cart        
    };    
}
const mapDispatchToProps = (dispatch) => ({    
    addToCartAction: (product) => dispatch(addToCartAction(product))
    });

export default connect(mapStateToProps, mapDispatchToProps)(CustList);

