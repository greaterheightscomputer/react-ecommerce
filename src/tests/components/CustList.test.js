import React from 'react';
import { shallow } from 'enzyme';
import { CustList } from '../../components/CustList';
import ecommerces from '../fixtures/ecommerces';

test('should render CustList with ecommerces', () => {
    const wrapper = shallow(<CustList ecommerces={ecommerces} />)
    expect(wrapper).toMatchSnapshot();
}); 

test('should render CustList with empty message', () => {
    const wrapper = shallow(<CustList ecommerces={[]} />)
    expect(wrapper).toMatchSnapshot();
}); 