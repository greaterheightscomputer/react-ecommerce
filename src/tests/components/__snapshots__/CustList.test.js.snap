// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CustList with ecommerces 1`] = `
<div>
  <h1>
    Customer List
  </h1>
  <CustListItem
    amount={12500}
    category="amen"
    createdAt={0}
    description="men shoe"
    id="1"
    image="shoe.jpeg"
    imageUrl=""
    item="men_footwear"
    key="1"
    stock={100}
  />
  <CustListItem
    amount={4500}
    category="women"
    createdAt={-432000000}
    description="women shoe"
    id="2"
    image="women_shoe.jpeg"
    imageUrl=""
    item="women_headwear"
    key="2"
    stock={403}
  />
  <CustListItem
    amount={350}
    category="boy"
    createdAt={432000000}
    description="boy shirt"
    id="3"
    image="shirt_boy.jpeg"
    imageUrl=""
    item="boy_clothe"
    key="3"
    stock={3200}
  />
  <CustListItem
    amount={2500}
    category="girl"
    createdAt={172800000}
    description="girl head tie"
    id="4"
    image="girl_headtie.jpeg"
    imageUrl=""
    item="girl_headwear"
    key="4"
    stock={17850}
  />
</div>
`;

exports[`should render CustList with empty message 1`] = `
<div>
  <h1>
    Customer List
  </h1>
</div>
`;
